app/models/ability.rb

```diff
@@ -38,16 +42,32 @@ def assign_staff_create_permission(current_user)
 
   def assign_staff_read_permission(current_user)
     can :read, EdiInvoice if /Y/.match?(current_user.edi_inv_view)
+    can :read, EdiErrorReport if /Y/.match?(current_user.edi_inv_view)
     can :read, Package if /A|Y/.match?(current_user.package_manage)
     can :read, PackageFile if /A|Y/.match?(current_user.package_manage)
+    can :read, Sal3BatchRequestsBc if /A|Y/.match?(current_user.sal3_batch_req)
   end
 
   def assign_staff_manage_permission(current_user)
     can :manage, AccessionNumberUpdate if /A|Y/.match?(current_user.accession_number)
     can :manage, EdiInvoice if /A|Y/.match?(current_user.edi_inv_manage)
     can :manage, EdiErrorReport if /A|Y/.match?(current_user.edi_inv_manage)
     can :manage, LobbytrackReport if /A|Y/.match? current_user.lobbytrack_report
-    can :manage, ManagementReport if /A|Y/.match?(current_user.mgt_rpts)
+  end
+
+  def assign_management_reports_permissions(current_user)
+    return unless /A|Y/.match?(current_user.mgt_rpts)
+
+    can :manage, CirculationStatisticsReport
+    can :manage, EdiInvLine
+    can :manage, EdiLin
```

Comment:
The EdiInvLine and EdiLin should probably be separated since authorized_user.edi_inv_manage should determine manage permissions for these and not authorized_user.mgt_rpts.